### PR TITLE
In Memory Lock as separate module

### LINKED ISSUE #805: Simple lock for tests
**Is your feature request related to a problem? Please describe.**
I would like to have a special lock that can be used in tests or for running single local instance

**Describe the solution you'd like**
I would like to have a simple in memory lock.

**Describe alternatives you've considered**
Using testcontainers with some DB can be an option but it takes more time for test to run. For simple tests I would like to have in memory solution.

### PR DESCRIPTION
**Pull request by kkocel**
Fixes #805

### CODE REVIEW

**Comment by lukas-krecan** [comment]
Hi, thanks, but I do not think this belongs to the core. It is not a production ready lock provider. Please just copy and paste the code if you need it, it's just few lines. 

We can document it, though.

**Comment by kkocel** [comment]
So maybe a separate module? It would be great to have a provider that can be easily used when running application locally on one machine.

**Comment by lukas-krecan** [comment]
If you are willing to create a spearate module, please do, that's something I will be happy to merge.

**Comment by kkocel** [comment]
@lukas-krecan I will create it throughout next week. I will add relevant Java doc to emphasize that it should be used only locally, for development and testing purposes. You mentioned that that provider is not production-ready - how it can be improved?

**Comment by lukas-krecan** [comment]
Cool. It's not production ready since it only works on one node. Other than that it should work.

**Comment by kkocel** [comment]
@lukas-krecan I refactored my PR to have the lock in the separate module. Can you take a look?

File: providers/inmemory/shedlock-provider-inmemory/pom.xml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            <activation>
+                <jdk>17</jdk>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <configuration>
+                            <argLine>--add-opens java.base/java.nio=ALL-UNNAMED --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.time=ALL-UNNAMED</argLine>
```
Comment (lukas-krecan) [inline providers/inmemory/shedlock-provider-inmemory/pom.xml:51]: Do we need this?

File: providers/inmemory/shedlock-provider-inmemory/src/main/java/net/javacrumbs/shedlock/provider/inmemory/InMemoryLockProvider.java [hunk 2]
```diff
@@ -36,7 +39,7 @@
         }
     }
 
-    boolean isLocked(String lockName) {
+    public boolean isLocked(String lockName) {
```
Comment (lukas-krecan) [inline providers/inmemory/shedlock-provider-inmemory/src/main/java/net/javacrumbs/shedlock/provider/inmemory/InMemoryLockProvider.java:42]: What is the reason for making this public?

**Review by lukas-krecan — COMMENTED** [review]
Thanks, looks good

File: providers/inmemory/shedlock-provider-inmemory/src/main/java/net/javacrumbs/shedlock/provider/inmemory/InMemoryLockProvider.java (hunk 2 above)
Comment (kkocel) [reply providers/inmemory/shedlock-provider-inmemory/src/main/java/net/javacrumbs/shedlock/provider/inmemory/InMemoryLockProvider.java:42]: different package for `InMemoryLockProviderIntegrationTest` - there is a cycle between support module and `InMemoryLockProvider`

File: providers/inmemory/shedlock-provider-inmemory/pom.xml (hunk 1 above)
Comment (kkocel) [reply providers/inmemory/shedlock-provider-inmemory/pom.xml]: removed

File: providers/inmemory/shedlock-provider-inmemory/src/main/java/net/javacrumbs/shedlock/provider/inmemory/InMemoryLockProvider.java (hunk 2 above)
Comment (kkocel) [reply providers/inmemory/shedlock-provider-inmemory/src/main/java/net/javacrumbs/shedlock/provider/inmemory/InMemoryLockProvider.java:42]: @lukas-krecan maybe you have better where to put tests for `InMemoryLockProvider` and avoiding maven module cycles?

Comment (lukas-krecan) [reply providers/inmemory/shedlock-provider-inmemory/src/main/java/net/javacrumbs/shedlock/provider/inmemory/InMemoryLockProvider.java:42]: Yes, I did this [cleanup](https://github.com/lukas-krecan/ShedLock/pull/810). I had to drop the KeepAlive test, but it was disabled anyways

**Comment by lukas-krecan** [comment]
Replaced by #810. Thanks

**Pull request closed without merging**